### DISCUSSION TITLE
Any plans for a new API database integration?

### DISCUSSION THREAD

**Discussion by MortalWombat-repo** [koreader/koreader#7991] [Q&A]
Sadly Goodreads is no more. Thats okay since Librarything is more extensive and probably more open. Unfortunately they locked their key this year. Does anyone know the management there and will there be a plan to migrate to that database?

Thank you.

**Comment by Frenzie** [comment]
That's probably more of a PRs welcome proposition. :)

**Reply by MortalWombat-repo** [reply]
> That's probably more of a PRs welcome proposition. :)
> […](#)
> On Mon, Jul 19, 2021, 11:40 Ico70 ***@***.***> wrote: Sadly Goodreads is no more. Thats okay since Librarything is more extensive and probably more open. Unfortunately they locked their key this year. Does anyone know the management there and will there be a plan to migrate to that database? Thank you. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#7991>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AABRQBMIF2UEWMOHQLBBWKTTYPXKTANCNFSM5ATKUQVA> .

Sadly not quite there yet, i hoped something was happening behind the scenes :)
If i manage to attempt it would goodreads commit be a good enough template
https://www.codeproject.com/Articles/472061/Goodreads-API-Tutorial, or do such projects need to be written from scratch? 
And im guessing nothing passes review if there isnt good security around it? From what i see most of you are professionals, and keep a high coding style standard.

**Reply by Frenzie** [reply]
If I'm not mistaken the LibraryThing API is more similar to Wallabag than to GoodReads (even if only on account of JSON vs XML; the data structure is likely to be slightly more similar between GR and LT). 

One could either take the GR plugin as a basis and go from there, cobble something together from parts of GR and Wallabag, or start completely from scratch. In any case the API output parsing in the old GR plugin is pretty hackish and should not be taken as an example unless strictly required.

Standards for inclusion in main are fairly strict, https://github.com/koreader/contrib is less strict. I.e., closer to "it works" than to "it should abide by our standards."
